File: rightSideView.py
class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

class Solution:
    def rightSideView(self, root: TreeNode) -> List[int]:
        res = []
        self.helper(root, 0, res)
        return res

    def helper(self, node: TreeNode, level: int, ans: List[int]) -> None:
        if not node:
            return

        if level == len(ans):
            ans.append(node.val)

        self.helper(node.right, level + 1, ans)
        self.helper(node.left, level + 1, ans)

File: rightSideView.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    void helper(TreeNode* node, int level, vector<int>&ans){
          if(node == NULL)return;

          if(level == ans.size())ans.push_back(node->val);

          helper(node->right, level+1, ans);
          helper(node->left, level+1, ans);
    }
    vector<int> rightSideView(TreeNode* root) {
          vector<int>res;
          helper(root, 0, res);

          return res;
    }
};